lib/Client.php

```diff
@@ -134,7 +132,7 @@ public function transaction() {
      */
     public function close() {
         /** @var Promise $promise */
-        $promise = all(promises($this->promisors));
+        $promise = all($this->promisors);
```

Comment:
Hmm ... with some distance since we initially did this I'm starting to wonder if that's not an unnecessary micro-optimization.  Would you mind convincing me otherwise, @bwoebi ?